Library/Homebrew/cmd/audit.rb

```diff
@@ -659,6 +659,28 @@ def audit_line(line, lineno)
     end
   end
 
+  def audit_prefix_has_contents
+    ps = formula.opt_prefix
+    a = "#{ps}/bin"
+    b = "#{ps}/lib"
+    c = "#{ps}/libexec"
+    d = "#{ps}/sbin"
+
+    if Dir["#{a}/*"].empty?
+      if Dir["#{b}/*"].empty?
+        if Dir["#{c}/*"].empty?
+          if Dir["#{d}/*"].empty?
+            problem <<-EOS.undent
+              The installation seems to be empty. Please ensure the prefix
+                 is set correctly and expected files are installed.
+                 The prefix configure/make argument may be case-sensitive.
+            EOS
+          end
+        end
+      end
+    end
+  end
```

Comment:
Yeah, still having problems. It seems to raise the error even when there's things present. I brewed the old version of ` osh `, which is what triggered this PR, and contains nothing in the install, then with:

```ruby

  def audit_prefix_has_contents
    return unless formula.prefix.directory?

    if Dir.glob("#{formula.prefix}/**/*").any? { |f| !(File.directory?(f) || Metafiles.copy?(f) || %w[.DS_Store INSTALL_RECEIPT.json].include?(File.basename(f))) }
      problem <<-EOS.undent
        The installation seems to be empty. Please ensure the prefix
           is set correctly and expected files are installed.
           The prefix configure/make argument may be case-sensitive.
      EOS
    end
  end
```

` brew audit osh ` got:

```
osh:
 * The installation seems to be empty. Please ensure the prefix
   is set correctly and expected files are installed.
   The prefix configure/make argument may be case-sensitive.
```

as expected, but doing ` mkdir lib && touch lib/test.dylib ` and then ` brew audit osh ` still gets the failure message.

```
tree
├── AUTHORS
├── CHANGES
├── INSTALL_RECEIPT.json
├── LICENSE
├── NOTES
├── README
└── lib
    └── test.dylib
```

```
osh:
 * The installation seems to be empty. Please ensure the prefix
   is set correctly and expected files are installed.
   The prefix configure/make argument may be case-sensitive.
```

Adding more files to various other expected directories still produces the error, including non-empty placeholder files.